server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -35,19 +37,32 @@ import com.cloudera.livy.server.batch.BatchSessionServlet
 import com.cloudera.livy.server.interactive.InteractiveSessionServlet
 import com.cloudera.livy.util.LineBufferedProcess
 
+
 class LivyServer extends Logging {
 
   private val ENVIRONMENT = LivyConf.Entry("livy.environment", "production")
   private val SERVER_HOST = LivyConf.Entry("livy.server.host", "0.0.0.0")
   private val SERVER_PORT = LivyConf.Entry("livy.server.port", 8998)
-  private val AUTH_TYPE = LivyConf.Entry("livy.server.auth.type", null)
-  private val KERBEROS_PRINCIPAL = LivyConf.Entry("livy.server.auth.kerberos.principal", null)
-  private val KERBEROS_KEYTAB = LivyConf.Entry("livy.server.auth.kerberos.keytab", null)
-  private val KERBEROS_NAME_RULES = LivyConf.Entry("livy.server.auth.kerberos.name_rules",
+  private val AUTH_KERBEROS_PRINCIPAL = LivyConf.Entry("livy.server.auth.kerberos.principal", null)
+  private val AUTH_KERBEROS_KEYTAB = LivyConf.Entry("livy.server.auth.kerberos.keytab", null)
+  private val AUTH_KERBEROS_NAME_RULES = LivyConf.Entry("livy.server.auth.kerberos.name_rules",
     "DEFAULT")
+  private val AUTH_TYPE = LivyConf.Entry("livy.server.auth.type", null)
+
+  private val LAUNCH_KERBEROS_PRINCIPAL =
+    LivyConf.Entry("livy.server.launch.kerberos.principal", null)
+  private val LAUNCH_KERBEROS_KEYTAB =
+    LivyConf.Entry("livy.server.launch.kerberos.keytab", null)
+  private val LAUNCH_KERBEROS_REFRESH_INTERVAL =
+    LivyConf.Entry("livy.server.launch.kerberos.refresh_interval_seconds", 3600)
+  private val KINIT_FAIL_THRESHOLD =
+      LivyConf.Entry("livy.server.launch.kerberos.kinit_fail_threshold", 5)
+  private val KINIT_CACHE_NAME =
+    LivyConf.Entry("livy.server.launch.kerberos.kinit_cache_name", "/tmp")
```

Comment:
This cannot be a directory. If it's a directory, `kinit` succeeds but doesn't write a ticket cache file, which is not what you want here.

Also, it would be better if instead of this, Livy created a temp directory to store temporary data. The location of the temp directory could be configured, and by default it can be just `java.io.tmpdir`. Then create a unique directory there and write temp data into it. Then you don't need a config entry for this.

Another thing that you're missing is exposing this to the spark-submit processes started by Livy. As the man page says, you need to set `KRB5CCNAME` in the process's environment.

As I mentioned in my original comment, it would be nice to do this, but you don't need to do it right now if you don't want to. But the code you added at the moment just won't work.